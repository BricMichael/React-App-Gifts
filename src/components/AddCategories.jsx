import React, { useState } from 'react';
import PropTypes from 'prop-types';


const AddCategories = ({ setCategories }) => {

    const [ input, setInput ] = useState('');

    const handleInput = ( e ) => {
        const event = e.target.value;
        setInput( event );

        console.log('el usuario me ha introducido valores ' + event) // asegurarme de que corre en los test
    }

    const handleButton = (/*e*/) => {
        if ( input.length > 2 ){
            setCategories( data => [ input, ...data ]) 
           // e.preventDefault();  no debe ir, quitar despues de las pruebas
        }
        // console.log('handleSubmit', input )  //pruebas
    }

    const handleSubmit = ( e ) => {
        e.preventDefault();
        e.target.reset();
    }


    return(
        <>
            <h2>SEARCH GIFS</h2>
            <form action="#" onSubmit = { handleSubmit }>
                <input 
                    type="text"
                    placeholder="Write..."
                    onChange = { handleInput }                
                />  
                <button onClick = { handleButton } >Search</button>
            </form>
        </>
    );
}

AddCategories.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategories;