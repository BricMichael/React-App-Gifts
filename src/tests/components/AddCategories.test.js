import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import AddCategories from "../../components/AddCategories";


describe('Pruebas en el componente <AddCategories />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategories setCategories = { setCategories }/>);

    beforeEach( () => {

        jest.clearAllMocks();
        wrapper = shallow(<AddCategories setCategories = { setCategories }/>)
    })

    test('Fotografiar el component AddCategories', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('probando el input del form', () => {
        const input = wrapper.find('input').simulate('change', { target: { value: 'pajarosSearch' } })

    });

     test('El submit no debe enviar la informacion', () => {
        const button = wrapper.find('button').simulate('click', { preventDefault(){} })

        expect( setCategories ).not.toHaveBeenCalled()

     })
    
    test('Debe llamar al setcategoires y limpiar la caja de texto', () => {
        const input = wrapper.find('input').simulate('change', { target: {value: 'Prueba exitosa'} })
    })


    test('el estado debe limpiarse', () => {
       
        expect(wrapper.find('input').prop('placeholder')).toBe('Write...')
    })

})
