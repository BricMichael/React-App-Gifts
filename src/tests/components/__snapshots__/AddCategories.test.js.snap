// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategories /> Fotografiar el component AddCategories 1`] = `
<Fragment>
  <h2>
    SEARCH GIFS
  </h2>
  <form
    action="#"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Write..."
      type="text"
    />
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </form>
</Fragment>
`;
